### PR TITLE
[TechDebt] Fix test coverage Tasking::construct_order_parameters

### PR DESCRIPTION
**Pull request by andher1802**
See changelog.

Checklist:
* [x] Test coverage
* [x] Version bump
* [x] Changelog update

### CODE REVIEW

File: tests/test_tasking.py [hunk 1]
```diff
@@ -21,17 +44,72 @@ def workspace():
...
 
-def test_construct_order_parameters(tasking_mock):
-    order_parameters = tasking_mock.construct_order_parameters(
-        data_product_id=constants.DATA_PRODUCT_ID,
-        name="my_tasking_order",
-        acquisition_start="2022-11-01",
-        acquisition_end="2022-11-10",
-        geometry=mock_search_parameters["intersects"],
+class TestTasking:
+    @pytest.fixture(autouse=True)
```
Comment (javidq) [inline tests/test_tasking.py:48]: autouse is not needed

File: up42/tasking.py [hunk 2]
```diff
@@ -80,31 +82,21 @@ def construct_order_parameters(
...
-            "dataProduct": data_product_id,
-            "params": {
-                "displayName": name,
-                "acquisitionStart": utils.format_time(acquisition_start),
-                "acquisitionEnd": utils.format_time(acquisition_end, set_end_of_day=True),
-            },
-        }
+        schema = self.get_data_product_schema(data_product_id)
+        params: dict[str, Any] = {param: None for param in schema["required"]}
+        params["displayName"] = name
```
Comment (javidq) [inline up42/tasking.py:87]:
```python
params |= {
            "displayName": name,
            "acquisitionStart": utils.format_time(acquisition_start),
            "acquisitionEnd": utils.format_time(acquisition_end, set_end_of_day=True),
        }
```

File: up42/tasking.py [hunk 3]
```diff
@@ -80,31 +82,21 @@ def construct_order_parameters(
...
-
-        schema = self.get_data_product_schema(data_product_id)
-        logger.info("See `tasking.get_data_product_schema(data_product_id)` for more detail on the parameter options.")
-        missing_params = {param: order_parameters["params"].get(param) for param in schema["required"]}
-        order_parameters["params"].update(missing_params)
-
         geometry = utils.any_vector_to_fc(vector=geometry)
         assert isinstance(order_parameters["params"], dict)
         if geometry["features"][0]["geometry"]["type"] == "Point":
-            # Tasking (e.g. Blacksky) can require Point geometry.
```
Comment (javidq) [inline up42/tasking.py:102]: Let's keep this for the moment, it provides context

File: up42/tasking.py [hunk 4]
```diff
@@ -80,31 +82,21 @@ def construct_order_parameters(
...
-                "acquisitionStart": utils.format_time(acquisition_start),
-                "acquisitionEnd": utils.format_time(acquisition_end, set_end_of_day=True),
-            },
-        }
+        schema = self.get_data_product_schema(data_product_id)
+        params: dict[str, Any] = {param: None for param in schema["required"]}
+        params["displayName"] = name
+        params["acquisitionStart"] = utils.format_time(acquisition_start)
+        params["acquisitionEnd"] = utils.format_time(acquisition_end, set_end_of_day=True)
+        order_parameters: order.OrderParams = {"dataProduct": data_product_id, "params": params}
```
Comment (javidq) [inline up42/tasking.py:90]: do we need typing hint here?

File: tests/test_tasking.py [hunk 5]
```diff
@@ -21,17 +44,72 @@ def workspace():
...
+        ],
+        ids=[
+            "tags:Value",
+            "tags:None",
+        ],
     )
-    assert isinstance(order_parameters, dict)
-    assert list(order_parameters.keys()) == ["dataProduct", "params"]
-    assert order_parameters["params"]["acquisitionMode"] is None
+    def test_construct_order_parameters(
```
Comment (javidq) [inline tests/test_tasking.py:71]: test_should_construct_order_parameters

File: tests/test_tasking.py [hunk 6]
```diff
@@ -21,17 +44,72 @@ def workspace():
...
+                        "type": "string",
+                        "title": "string",
+                        "format": "string",
+                    }
+                },
+            },
+        )
+        order_parameters = tasking_obj.construct_order_parameters(
+            data_product_id=constants.DATA_PRODUCT_ID,
+            name="some-name",
```
Comment (javidq) [inline tests/test_tasking.py:95]: let's extract this to a variable or constant since it's longer than 5 symbols

File: tests/test_tasking.py [hunk 7]
```diff
@@ -21,17 +44,72 @@ def workspace():
...
+        ],
     )
-    assert isinstance(order_parameters, dict)
-    assert list(order_parameters.keys()) == ["dataProduct", "params"]
-    assert order_parameters["params"]["acquisitionMode"] is None
+    def test_construct_order_parameters(
+        self,
+        requests_mock: req_mock.Mocker,
+        tasking_obj: tasking.Tasking,
+        input_geometry: Optional[tasking.Geometry],
```
Comment (javidq) [inline tests/test_tasking.py:75]: it's not optional

File: tests/test_tasking.py [hunk 8]
```diff
@@ -21,17 +44,72 @@ def workspace():
...
+            acquisition_start=START_DATE,
+            acquisition_end=END_DATE,
+            geometry=input_geometry,
+            tags=tags,
+        )
+        expected = {
+            "params": {
+                "displayName": "some-name",
+                "acquisitionStart": "2014-01-01T00:00:00Z",
+                "acquisitionEnd": "2022-12-31T23:59:59Z",
```
Comment (javidq) [inline tests/test_tasking.py:105]: These two values should not be hard coded in the test, to be extracted to constants

**Review by javidq — CHANGES_REQUESTED** [review]

File: up42/tasking.py [hunk 9]
```diff
@@ -14,6 +14,16 @@
 
 logger = utils.get_logger(__name__)
 
+Geometry = Union[
```
[the remark this answers is not in the document]
Comment (andher1802) [reply up42/tasking.py:17]: It is different

```
Geometry = Union[
    dict,
    geojson.Feature,
    geojson.FeatureCollection,
    list,
    geopandas.GeoDataFrame,
    geom.Polygon,
]
```

File: up42/tasking.py (hunk 4 above)
Comment (andher1802) [reply up42/tasking.py:90]: If we don't add type, wouldn't be a dict type.

**Review by javidq — APPROVED** [review]

**Pull request merged**